src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt

```diff
@@ -0,0 +1,56 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.jooq.RecordMapper
+import org.jooq.Record
+
+class OwnerDao(private val sql: DSLContext) {
+    private val owner = OwnerTable.instance
+    private val pet = PetTable.instance
+
+    private val ownerDataMapper = RecordMapper<Record,OwnerData>{ record ->
+            OwnerData(
+                record[owner.id],
+                record[owner.ownerName],
+                record[owner.employeeId],
+                record[owner.companyId]
+            )
+    }
+
+    //Create a function that retrieve all the owners from the owner table and return their name, employee Id and company Id
+    fun getAllOwners(companyId: Long): List<OwnerData>{
+        return sql.select(owner.id, owner.ownerName,owner.employeeId,owner.companyId)
+            .from(owner)
+            .where(owner.companyId.eq(companyId))
+            .fetch(ownerDataMapper)
+    }
+
+//    Add a new API that get pet id and  return the owner’s information.
+    fun getOwnerByPetId(petId: Long, companyId: Long): OwnerData?{
+        return sql.select(owner.id, owner.ownerName,owner.employeeId,owner.companyId)
+            .from(pet)
+            .join(owner).on(pet.ownerId.eq(owner.id))
```

Comment:
Do we want this join operation or different join?
What are the difference between all the join operations?

Reply:
Thanks for the input! You’re right, I’ve switched to a RIGHT JOIN to ensure we get all rows from the owner table, even if the pet isn’t linked to an owner.